### PR TITLE
fix: address slow shutdowns in aggregation loop

### PR DESCRIPTION
**Pull request by MSevey**
## Overview

I started trying to debug the test failure in  #1058 and #1095 by first cleaning up the test output. I replaced the stdout logger with the file logger and that highlighted some panic conditions of writing to a closed file, which meant there were some hanging threads. The test that was easiest to catch the panic was `TestTx`.

One of the issues was the indexer didn't have a ctx so wasn't shutting down cleanly.

The second was the edge case of the aggregation loop execution taking some time, resulting in the node shutting down in the middle. I added the ctx check at the end so that any updates are ideally saved to the store before we exit. 

I tested this locally by running the failing test in a loop. Prior to these changes it panicked every time, afterwards, you can run the test in a loop without failure. 

## Summary by CodeRabbit

Refactor:
- Enhanced the node shutdown process in `block/manager.go` to skip publishing if the node is already shut down, improving system efficiency.
- Modified the `createAndStartIndexerService` function in `node/full.go` to require a context during initialization, allowing for better control over the service's lifecycle.

Tests:
- Updated `node/full_client_test.go` to use a cancellable context and a new file logger, improving test reliability and logging capabilities.
- Modified `state/txindex/indexer_service_test.go` to use a cancellable context, ensuring proper cleanup after tests.

Style:
- Updated `state/txindex/indexer_service.go` to include a context in the `IndexerService` struct and handle its cancellation, aligning with the context-oriented design.

### CODE REVIEW

**Review by Manav-Aggarwal — DISMISSED** [review]
made one change moving context to front of indexer service, LGTM

**Review by Manav-Aggarwal — APPROVED** [review]

File: block/manager.go [hunk 1]
```diff
@@ -699,6 +699,13 @@ func (m *Manager) publishBlock(ctx context.Context) error {
 		return err
 	}
 
+	// Check if the node has shutdown prior to publishing to channels
+	select {
+	case <-ctx.Done():
+		return nil
```
Comment (gupadhyaya) [inline block/manager.go:705]: since `nil` indicates successful scenario, node shutting down before publishing the block should return error instead or `nil` is still fine?

Comment (MSevey) [reply block/manager.go:705]: I think you can look at it either way. 

The only actions after this step are signaling background loops. So at this point things have been successful. Deciding whether to quit early because of a shutdown, vs blocking on signaling background loops that are also shutting down seems like a successful scenario to me.

**Review by gupadhyaya — APPROVED** [review]

**Review by Manav-Aggarwal — APPROVED** [review]
